butterfly/QueryLayer/Query.py

```diff
@@ -44,6 +44,14 @@ def update_source(self, keywords):
             output.SIZE.Y.NAME: int(Y),
             output.SIZE.X.NAME: int(X)
         }
+        # Get the right kind of datsource
+        runtime.SOURCE.VALUE = keywords[runtime.SOURCE.NAME]
+        # Optional keywords by source
+        inH5 = runtime.SOURCE.HDF5.INNER
+        optional_fields = [inH5, output.PATH]
+        # Assign all optional keywords
+        for op in optional_fields:
+            op.VALUE = keywords.get(op.NAME,op.VALUE)
 
```

Comment:
There should definitely be an error if there is no source value. 
There should be different errors for no block, type, or size.